src/frontends/tensorflow/src/input_model.cpp

```diff
@@ -195,6 +195,16 @@ std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::determine_cu
         if (ops_done.count(current_operation_place) == 0) {
             bool can_add = true;
             auto input_count = current_operation_decoder->get_input_size();
+            const auto& curr_op_type = current_operation_decoder->get_op_type();
+
+            // break the cycle
```

Comment:
don't understand the reason of this cycle
could you explain in more detail?

could you add some tests?

Reply:
In case presence of `NextIteration` node, the should be a cycle in the graph that we need to break. Currently, TF1 while is not supported. This change is to avoid cycling in execution.

Yes, I can add test.

Reply:
Can it be solved by verifying that op is already in `ops_to_do`? What if not only `NextIteration` can create cycle?

Reply:
@mvafin, I was thinking about the same idea, even didn't look at GitHub:) I did it.